sda-auth/info.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/kataras/iris/v12"
+	"github.com/neicnordic/crypt4gh/keys"
+	log "github.com/sirupsen/logrus"
+)
+
+type Info struct {
+	ClientID  string `json:"client_id"`
+	OidcURI   string `json:"oidc_uri"`
+	PublicKey string `json:"public_key"`
+	InboxURI  string `json:"inbox_uri"`
+}
+
+func readPublicKeyFile(filename string) (key *[32]byte, err error) {
+	log.Info("Reading Public key file")
+	file, err := os.Open(filepath.Clean(filename))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+	publicKey, err := keys.ReadPublicKey(file)
+	if err != nil {
+		return nil, fmt.Errorf("error while reading public key file %s: %v", filename, err)
+	}
+
+	return &publicKey, err
+}
+
+func (auth AuthHandler) getInfo(ctx iris.Context) {
+	publicKey, err := readPublicKeyFile(auth.Config.C4ghPubKey)
```

Comment:
Not a huge thing, but could we cache this without making things ugly?

Reply:
sure! let me take a look

Reply:
@pontus @jbygdell  Do you have a suggestion for the caching from another repo?
If we are already using a lib I think I should probably use that or introduce an explicit implementation instead.
I've only found this one being used, but it doesn't look very well maintained:
[github.com/dgraph-io/ristretto](github.com/dgraph-io/ristretto)

Reply:
Sorry if it was unclear, this doesn't have to/shouldn't be a cache. It's just one value that should be available (or not, if allowed without it).

I believe you can just stick the public key in `AuthHandler`.